src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1076,7 +1076,9 @@ private static string GenerateForType(TypeGenerationSpec typeMetadata, string me
     get => _{typeFriendlyName} ??= {typeMetadata.CreateTypeInfoMethodName}({OptionsInstanceVariableName});
 }}
 
-private static {typeInfoPropertyTypeRef} {typeMetadata.CreateTypeInfoMethodName}({JsonSerializerOptionsTypeRef} {OptionsLocalVariableName})
+// Intentionally not a static method because we create a delegate to it. Invoking delegates to instance
```

Comment:
This comment will be visible to users in generated code, and it might not be very relevant to them. I would recommend moving this to the source generator implementation instead (e.g., line 1072).

Reply:
Honestly I think that is fine in case users will inspect source code and wonder why we didn't make it static